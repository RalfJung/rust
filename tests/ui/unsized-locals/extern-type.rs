// build-fail
#![feature(extern_types)]
#![feature(unsized_fn_params)]
#![crate_type = "lib"]

extern {
    pub type E;
}

fn test(e: E) {} //~ERROR: does not have a dynamically computable size

pub fn calltest(e: Box<E>) {
    test(*e) //~ERROR: does not have a dynamically computable size
}
