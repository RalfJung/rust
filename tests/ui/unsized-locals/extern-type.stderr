error: function argument does not have a dynamically computable size
  --> $DIR/extern-type.rs:10:9
   |
LL | fn test(e: E) {}
   |         ^
   |
   = note: `E` contains an `extern type`, which is not allowed in function arguments

error: function parameter does not have a dynamically computable size
  --> $DIR/extern-type.rs:13:5
   |
LL |     test(*e)
   |     ^^^^^^^^
   |
   = note: `E` contains an `extern type`, which is not allowed in function parameters

error: aborting due to 2 previous errors

